src/Parser/Template.cs

```diff
@@ -54,6 +57,95 @@ public Template(string template, bool isPositional, List<Literal> literals, List
       Holes = holes.ToArray();
     }
 
+    public string Render(object[] parameters)
```

Comment:
> I am a bit unsettled by the fact that all the members of Template would need to be internal for a renderer class to work.

Well or just public?



> The one thing I would drop when we're done is the Rebuild method, which in my opinion is not required anymore. We will have other means to test the template.

I like to keep it. Makes it easier to find an issue (if it's a parse or render issue). We could move it to partial/new class.